pkg/operator/resources/cluster/crd.go

```diff
@@ -0,0 +1,160 @@
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "datavolumes.cdi.kubevirt.io",
+			Labels: map[string]string{
+				"cdikubevirt.io": "",
+			},
+		},
+		Spec: extv1beta1.CustomResourceDefinitionSpec{
+			Group: "cdi.kubevirt.io",
+			Names: extv1beta1.CustomResourceDefinitionNames{
+				Kind:   "DataVolume",
+				Plural: "datavolumes",
+				ShortNames: []string{
+					"dv",
+					"dvs",
+				},
+				Singular: "datavolume",
+			},
+			Version: "v1alpha1",
+			Scope:   "Namespaced",
+			Validation: &extv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &extv1beta1.JSONSchemaProps{
+					Properties: map[string]extv1beta1.JSONSchemaProps{
+						"apiVersion": extv1beta1.JSONSchemaProps{
+							Type: "string",
+						},
+						"kind": extv1beta1.JSONSchemaProps{
+							Type: "string",
+						},
+						"metadata": extv1beta1.JSONSchemaProps{},
+						"spec": extv1beta1.JSONSchemaProps{
+							Properties: map[string]extv1beta1.JSONSchemaProps{
+								"source": extv1beta1.JSONSchemaProps{
+									Properties: map[string]extv1beta1.JSONSchemaProps{
+										"http": extv1beta1.JSONSchemaProps{
+											Properties: map[string]extv1beta1.JSONSchemaProps{
+												"url": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+												"secretRef": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+											},
+											Required: []string{
+												"url",
+											},
+										},
+										"s3": extv1beta1.JSONSchemaProps{
+											Properties: map[string]extv1beta1.JSONSchemaProps{
+												"url": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+												"secretRef": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+											},
+											Required: []string{
+												"url",
+											},
+										},
+										"registry": extv1beta1.JSONSchemaProps{
+											Properties: map[string]extv1beta1.JSONSchemaProps{
+												"url": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+												"secretRef": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+											},
+											Required: []string{
+												"url",
+											},
+										},
+										"pvc": extv1beta1.JSONSchemaProps{
+											Properties: map[string]extv1beta1.JSONSchemaProps{
+												"namespace": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+												"name": extv1beta1.JSONSchemaProps{
+													Type: "string",
+												},
+											},
+											Required: []string{
+												"namespace",
+												"name",
+											},
+										},
+										"upload": extv1beta1.JSONSchemaProps{},
+										"blank":  extv1beta1.JSONSchemaProps{},
+									},
+								},
+								"pvc": extv1beta1.JSONSchemaProps{
+									Properties: map[string]extv1beta1.JSONSchemaProps{
+										"resources": extv1beta1.JSONSchemaProps{
```

Comment:
Is there a reason that we are not using the real PVC types in here?

Reply:
This is open api validation stuff which works at the json level.  Maybe @igoihman can provide more context